### DISCUSSION TITLE
Appshell: Is it possible to push footer below entirety of page content?

### DISCUSSION THREAD

**Discussion by khari998** [skeletonlabs/skeleton#1406] [Q&A] [answered]
I am using the AppShell in my layout and made a header and footer. 

Now when I am making content between them, the footer seems to display as if it is fixed and does not render below the entirety of the content.  There is a scrollbar for only the test content when I would want the scrollbar to extend the entirety of the page.

I followed all instructions and added the required classes to the body and global postcss. 

Screenshots are  from an app created with `npm create skeleton-app@latest my-skeleton-app`

<img width="1248" alt="Screenshot 2023-04-30 at 12 55 54 PM" src="https://user-images.githubusercontent.com/40578958/235365934-62966de4-d972-4d51-84da-d1bb9a98d6d7.png">

<img width="1246" alt="Screenshot 2023-04-30 at 12 52 04 PM" src="https://user-images.githubusercontent.com/40578958/235365739-395867a5-7831-429e-9627-251fba9c6261.png">

Ideally the test content would take up the rest of the screen and the footer wouldn't be seen initially. The footer should only be seen when the end of the test content is reached after scrolling through the test content. 

(Excuse the terrible colors, this was only for highlighting the layout problem I'm having)

**Comment by khari998** [comment]
I figured out this is possible by breaking my Footer component out of the footer slot in the Appshell. 

Not ideal but it works as expected now.

**Comment by Sarenor** [comment]
We sadly have 0 information to help you right now.
Could you provide the code you're using, I think what you want is very easily possible to do with the AppShell

**Comment by khari998** [comment]
Sure thing.

`src/routes/+layout.svelte`:

```svelte
<script lang='ts'>
	// The ordering of these imports is critical to your app working properly
	import '@skeletonlabs/skeleton/themes/theme-skeleton.css';
	// If you have source.organizeImports set to true in VSCode, then it will auto change this ordering
	import '@skeletonlabs/skeleton/styles/all.css';
	// Most of your app wide CSS should be put in this file
	import '../app.postcss';
	import { AppShell } from '@skeletonlabs/skeleton';

</script>

<AppShell>
	<svelte:fragment slot="header">Header</svelte:fragment>
	<slot></slot>
	<svelte:fragment slot="footer">
		<div class="h-36">
			Footer
		</div>
	</svelte:fragment>
</AppShell>
```

`src/routes/+page.svelte`:

```svelte
<div class="h-screen flex flex-col  justify-between bg-red-600">
    <div>
        Beginning of Test content

    </div>

    <div>
        End of Test content
    </div>
</div>
```

`src/app.postcss`:

```css
/*place global styles here */
html,
body {
	@apply h-full overflow-hidden;
}
```

`src/app.html`:

```html
<!DOCTYPE html>
<html lang="en" class="dark">
	<head>
		<meta charset="utf-8" />
		<link rel="icon" href="%sveltekit.assets%/favicon.png" />
		<meta name="viewport" content="width=device-width" />
		%sveltekit.head%
	</head>
	<body>
		<div style="display: contents" class="h-full overflow-hidden">%sveltekit.body%</div>
	</body>
</html>
```

**Comment by khari998** [comment, marked as answer]
Finally figured out what I was doing wrong. I was using the Header and Footer slots instead of PageHeader and PageFooter slots. The difference between the two wasn't clear until I messed around with it. After switching, everything works as expected now.

**Reply by Sarenor** [reply]
Sorry, I didn't see the first answer. Nice that that worked!

Any suggestions on how to make the distinction more clear?

**Reply by khari998** [reply]
I would just put a note in the documentation about the positioning of each. It seems as if Header and Footer use fixed positioning while PageHeader and PageFooter use relative positioning (I don't know exactly how they are positioned by default in the Appshell component because they seem like they are grouped outside of the page) It helps with understanding when to use either. The Header and Footer slots don't scroll with the page while the PageHeader and PageFooter slots do scroll with the page.

**Reply by endigo9740** [reply]
This message already exists on the documentation, along with a visualization.

![image](https://user-images.githubusercontent.com/1509726/235377238-4ec7d6fe-6675-4d9c-96f3-a30ab4ab1aaa.png)

**Reply by khari998** [reply]
I must've misunderstood it but it makes sense more sense on a second look. Thank you!
